### DISCUSSION TITLE
Heat Thoughts

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#521] [Ideas]
This is some assorted thoughts, there might be gaps or missing ideas. Call me out if something doesn't make sense.

---

Ultimately heat transfer is going to be a bit arbitrary, which is fine. We don't have to simulate the universe perfectly, but we do have to have _some_ kind of rules for how things work. In the case of heat, we're basing it off of real equations but fudging the numbers so we don't have to define every property of every thing.
 
### Heat Radiation

Systems automatically radiate excess heat into space. This resource is great for figuring out how all of this works from a hard science perspective.

http://www.projectrho.com/public_html/rocket/heatrad.php

Heat is radiated out based on this simplified equation:

`P = A * σ * T^4`

Where `P` is watts, `A` is the area of the radiator, `σ` is the [Stefan-Boltzmann constant](https://en.wikipedia.org/wiki/Stefan–Boltzmann_constant) and `T^4` is the temperature of the radiator raised to the fourth power. This is way too little dissipation for the power we're dealing with, and very boring in a sci-fi setting. Therefore, I'm adjusting the formula just a little bit.

`P = A * a * T^5`

We can hand-wave the explanation for why it's raised to the 5th as using some kind of warp field to change the speed of light value used in the Stefan-Boltzmann constant.

We'll assume each heat radiator is 1 meter square, which makes practical sense and should be sufficient.

### Transferring Heat

Heat will be transferred from systems to coolant, and then from coolant to radiators. Systems need sufficient coolant in them in order to transfer as much heat as possible.

We can use this formula for converting heat into watts and vice versa:

`(J / specific heat capacity) / mass (of the heated object)`

> `J` = 1 wattsecond

Here's the specific heat capacity of possible mediums for transferring heat, in J/gK:
- Ammonia Gas: 2.061
- Ammonia Liquid: 4.7
- Water Vapor: 1.865
- Water Liquid: 4.18
- Steel: .475
- Titanium: .52
- Aluminum: .904

Specific heat requires a mass value and a temperature value on the item that heat is transferring through, but we only track inventory temperature, not mass. I don't really want to make it possible to assign mass to inventory - that seems a bit excessive - so for inventory we'll just assume everything has the density of `1 g / cm^3`, which is about the density of water. This works well because we're using liquid coolant to transfer heat to the radiators anyway.

For ship systems, we don't really have a volume, so a handwavey density doesn't work. Instead we can just say all systems have the same mass. 1 metric ton (1000 kg) sounds good to me, but it might be worth adjusting in the future.

### How it works

- Systems generate heat, which instantly heats up the system.
- That system's heat will transfer slowly from the system to the coolant in the room. 
- Any coolant that is inside of a system room is assumed to be attached to a radiator, and will radiate heat.
- Any coolant that is not inside a system room will not radiate heat, and will retain that heat perfectly.

**Comment by tannerchamb** [comment]
I think it could be useful to have imaginary materials like Tritanium or Plasma Coolant that have made up or better specific heat capacities if we need them.

I'd also like to add the idea of heat sinks, which are one time use radiators that are dumped full of heat and ejected out of the ship. They could be used for dumping excess heat from running the phasers a bit long, during charging a superweapon, or flying near a star or nebula. Or if stealth had a heat component, closing the radiator vents to lower the ship's sensor signature could be possible if the heat gets shunted into a heatsink.

**Reply by alexanderson1993** [reply]
I think the way I'm modeling this will make those imaginary materials a possibility.

The heat sink you describe is totally possible - it's basically a coolant item that you store away from a system, so it can't gain or lose any heat. We just need to add the ability to eject inventory.
